### PR TITLE
M3-1503 Implement Pagey on Volumes Landing

### PR DESCRIPTION
**Pull request by martinmckenna**
### Purpose

Use Pagey HOC on VolumesLanding instead of tracking pagination inside the component

### Notes

* Created a `WithEvents` HOC component to handle injecting the event data into the updating Volume.
   * Take special notice of the `data` prop passed to it's child. It's either going to be the plain old data it gets from `<Pagey />` or it will pass down augmented data if an event has occurred on a Volume

### Todo
- [ ] Sorting table headers, but I wanted feedback on this PR first

### CODE REVIEW

File: src/features/Volumes/VolumesLanding.tsx [hunk 1]
```diff
@@ -167,37 +164,24 @@ class VolumesLanding extends React.Component<CombinedProps, State> {
       ))
       .merge(updateVolumes$)
       .subscribe((event) => {
-        this.getVolumes()
-          .then((volumes) => {
-            if (!volumes || !this.mounted) { return; }
+        this.props.request()
```
Comment (rjstires) [inline src/features/Volumes/VolumesLanding.tsx]: What happens if the Volume for which the event occurred is on page 7, and we're viewing page 1?

File: src/features/Volumes/VolumesLanding.tsx [hunk 2]
```diff
@@ -167,37 +164,24 @@ class VolumesLanding extends React.Component<CombinedProps, State> {
       ))
       .merge(updateVolumes$)
```
Comment (rjstires) [inline src/features/Volumes/VolumesLanding.tsx]: Is this still needed?

**Comment by Jskobos** [comment]
I'm not quite sure what volumes is doing in VolumesLanding state. It isn't read anywhere in the file; if it needs to be maintained, could it be in the `withEvents` state? Or is this due to reusability concerns?

Also, same thing with `eventsSub`; it's declared in the landing page, but only referred to from within the HOC. Is there any reason it can't be declared in there, or is it a `this` issue?

**Comment by martinmckenna** [comment]
@Jskobos That was just an oversight. I removed the unused code

**Pull request merged**